gapir/cc/resource_loader.h

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (C) 2018 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef GAPIR_RESOURCE_LOADER_H
+#define GAPIR_RESOURCE_LOADER_H
+
+#include "replay_service.h"
+#include "resource.h"
+
+#include <stdint.h>
+#include <memory>
+
+namespace gapir {
+
+// RessourceLoader is an interface which can load a list of resources in-orderly
+// to the specified location.
+// TODO(qining): Change the load() or fetch() interface to accept a callback
+// function to process the fetched data, then we won't need two methods anymore.
+class ResourceLoader {
+ public:
+  virtual ~ResourceLoader() {}
+
+  // Loads count resources from the provider and writes them, in-order, to
+  // target. If the net size of all the resources exceeds size, then false is
+  // returned.
+  virtual bool load(const Resource* resources, size_t count, void* target,
+                    size_t targetSize) = 0;
+
+  // Fetch queries the specified resources and returns a
+  // ReplayService::Resources instance which contains the resources data.
+  virtual std::unique_ptr<ReplayService::Resources> fetch(
+      const Resource* resources, size_t count) = 0;
+};
+
+// SimpleResourceLoader implements the ResourceLoader interface. It pull
+// resources from a ReplayService instance for every resource loading request.
+class SimpleResourceLoader : public ResourceLoader {
```

Comment:
I would prefer a name other than "Simple". That doesn't tell us how it's simple.

PassthroughResourceLoader may be appropriate.